mage_ai/cache/block.py

```diff
@@ -62,10 +71,13 @@ def build_key(self, block: Union[Dict], repo_path: str = None) -> str:
             configuration = block.configuration or {}
             file_path = block.file_path
 
-        if configuration:
-            file_source = (configuration or {}).get('file_source') or {}
-            if file_source and (file_source or {}).get('path'):
+        file_source = (configuration or {}).get('file_source') or {}
+        if file_source:
+            if (file_source or {}).get('path'):
                 return remove_base_repo_path_or_name((file_source or {}).get('path'))
+        elif block_language:
+            block_file_extension = f'.{BLOCK_LANGUAGE_TO_FILE_EXTENSION[block_language]}'
+            file_path = f'{file_path}{block_file_extension}'
```

Comment:
i don't fully understand what's going on here, but is there a chance the original `file_path` has the file extension already?

Reply:
When the block is a dictionary (which when testing was when the block cache is initialized), the `file_path` does not include the file extension, but when the block is a `Block` model, the `file_path` does include the file extension already.

So that's why the `block_language` variable is only reassigned in the `if isinstance(block, dict)` section. I actually initially included the `block_language` in the `else` block, but then the block cache key had a double file extension on the end.

Reply:
If the block is a dictionary, I don't think the `file_path` would ever have the file extension since it's using this code to build it and the `block_uuid` doesn't have a file extension: 
```
file_path = os.path.join(
                repo_path,
                block_type_plural,
                block_uuid,
            )
```